src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CourtCaseMapper.java

```diff
@@ -32,16 +41,129 @@ public static CourtCaseEntity create(CourtCaseEntity courtCaseEntity, String upd
             .sessionStartTime(courtCaseEntity.getSessionStartTime())
             .sourceType(courtCaseEntity.getSourceType())
             .suspendedSentenceOrder(courtCaseEntity.getSuspendedSentenceOrder())
+            .hearings(courtCaseEntity.getHearings()
+                            .stream()
+                            .map(CourtCaseMapper::createHearing)
+                            .collect(Collectors.toList()))
             .offences(courtCaseEntity.getOffences()
                             .stream()
                             .map(CourtCaseMapper::createOffence)
                             .collect(Collectors.toList()))
+            .defendants(Optional.ofNullable(courtCaseEntity.getDefendants()).orElse(Collections.emptyList())
+                .stream()
+                .map(defendant -> createDefendant(defendant, crn, updatedProbationStatus))
+                .collect(Collectors.toList()))
             .build();
 
         newCourtCaseEntity.getOffences().forEach(offenceEntity -> offenceEntity.setCourtCase(newCourtCaseEntity));
+        newCourtCaseEntity.getDefendants().forEach(defendantEntity -> defendantEntity.setCourtCase(courtCaseEntity));
+        newCourtCaseEntity.getHearings().forEach(hearingEntity -> hearingEntity.setCourtCase(courtCaseEntity));
         return newCourtCaseEntity;
     }
 
+    public static CourtCaseEntity mergeDefendantsOnCase(CourtCaseEntity existingCase, CourtCaseEntity updatedCase, String defendantId) {
+        final var existingDefendants = Optional.ofNullable(existingCase.getDefendants()).orElse(Collections.emptyList());
+        final var existingHearings = CourtCaseMapper.createHearings(existingCase.getHearings());
+        if (existingDefendants.size() <= 1) {
+            final var caseToSave = updatedCase.withHearings(existingHearings);
+            applyParentToCollections(caseToSave);
+            return caseToSave;
+        }
+
+        final var updatedDefendant = updatedCase.getDefendants().stream()
+            .filter(existingDef -> defendantId.equalsIgnoreCase(existingDef.getDefendantId()))
+            .findFirst()
+            .orElseThrow(() -> new ConflictingInputException(String.format("No defendant ID matching %s in updated case", defendantId)));
+
+        // Clone all OTHER defendants, removing the IDs
```

Comment:
OK - will do but that is nothing new in CourtCaseMapper. It was like that when I looked at it for the update of the probation status and there were no comments then. So yes, when I came to this, I believed that the ids are dropped so that records are created anew.